packages/context/test/unit/resolver.unit.ts

```diff
@@ -595,7 +596,7 @@ function customAsyncDecorator(def: Object) {
   return inject('foo', def, async (c: Context, injection: Injection) => {
     const barKey = injection.metadata && injection.metadata.x;
     const b = await c.get(barKey);
-    const f = await c.get(injection.bindingKey);
+    const f = await c.get(injection.bindingSelector as BindingAddress);
```

Comment:
I think it makes sense to modify `ctx.get` and `ctx.getAsync` to work support binding filters too. What do you think?

This is the API I have in my mind:

```ts
get<T>(key: BindingAddress<T>): Promise<T>;
get<T>(filter: BindingFilter<T>): Promise<T[]>;
```

But this will not help us here, because `bindingSelector` can be either type and thus the compiler will think that the result of `ctx.get` can either an array or a value and the compiler won't be able to pick the right one only. Hmm :(

Let's leave this out of scope of this pull request.

Reply:
> I think it makes sense to modify ctx.get and ctx.getAsync to work support binding filters too. What do you think?

Maybe not, since we already have `ctx.find` that takes `BindingFilter`.